source/code/scxsystemlib/common/scxsysteminfo.cpp

```diff
@@ -52,6 +52,10 @@
 
 const std::wstring s_defaultSudoPath = L"/etc/opt/microsoft/scx/conf/sudodir/sudo";
 
+// Keeping the tmp dir location here so that in future if any enhancements are made 
+// that can be placed in this class.
+const std::string s_defaultTmpDir = "/etc/opt/microsoft/scx/conf/tmpdir/";
```

Comment:
PAL is common for multiple products. You can't have hard-coded strings like this, please let the product set it by calling SetDefaultTemporaryDirectory(). That way, any products that use the PAL can have appropriate locations on a product-specific basis.

See how logging customization was handled; you can set logging locations, logging headers, log format, etc, on a per-project basis.